### PR TITLE
SQSERVICES-547 add feature config for guest links

### PR DESCRIPTION
**Pull request by battermann**
Implementation for https://wearezeta.atlassian.net/browse/SQSERVICES-1045 and https://wearezeta.atlassian.net/browse/SQSERVICES-1046 for [Epic](https://wearezeta.atlassian.net/browse/SQSERVICES-547)

## Checklist

 - [x] The **PR Title** explains the impact of the change.
 - [x] The **PR description** provides context as to why the change should occur and what the code contributes to that effect. This could also be a link to a JIRA ticket or a Github issue, if there is one.
 - [x] If HTTP endpoint paths have been added or renamed, the **endpoint / config-flag checklist** (see Wire-employee only backend [wiki page](https://github.com/zinfra/backend-wiki/wiki/Checklists)) has been followed.
 - [x] If a cassandra schema migration has been added, I ran **`make git-add-cassandra-schema`** to update the cassandra schema documentation.
 - [x] **changelog.d** contains the following bits of information ([details](https://github.com/wireapp/wire-server/blob/develop/docs/developer/changelog.md)):
   - [x] A file with the changelog entry in one or more suitable sub-sections. The sub-sections are marked by directories inside `changelog.d`.
   - [x] If new config options introduced: added usage description under docs/reference/config-options.md
   - [x] If new config options introduced: recommended measures to be taken by on-premise instance operators.
   - [x] If a cassandra schema migration is backwards incompatible (see also [these docs](https://github.com/wireapp/wire-server/blob/develop/docs/developer/cassandra-interaction.md#cassandra-schema-migrations)), measures to be taken by on-premise instance operators are explained.
   - [x] If a data migration (not schema migration) introduced: measures to be taken by on-premise instance operators.
   - [x] If public end-points have been changed or added: does nginz need un upgrade?
   - [x] If internal end-points have been added or changed: which services have to be deployed in a specific order?

### CODE REVIEW

File: services/galley/src/Galley/Cassandra/TeamFeatures.hs [hunk 1]
```diff
@@ -190,8 +211,9 @@ interpretTeamFeatureStoreToCassandra ::
   Sem (TeamFeatureStore ': r) a ->
   Sem r a
 interpretTeamFeatureStoreToCassandra = interpret $ \case
-  GetFeatureStatusNoConfig' ps tfn tid -> embedClient $ getFeatureStatusNoConfig ps tfn tid
-  SetFeatureStatusNoConfig' ps tfn tid value -> embedClient $ setFeatureStatusNoConfig ps tfn tid value
+  GetFeatureStatusNoConfig' p tid -> embedClient $ getFeatureStatusNoConfig p tid
+  GetFeatureStatusNoConfigAndLockStatus' p tid -> embedClient $ getFeatureStatusNoConfigAndLockStatus p tid
+  SetFeatureStatusNoConfig' p tid value -> embedClient $ setFeatureStatusNoConfig p tid value
```
Comment (fisx) [inline services/galley/src/Galley/Cassandra/TeamFeatures.hs:216]:
```suggestion
  GetFeatureStatusNoConfig' tfn tid -> embedClient $ getFeatureStatusNoConfig tfn tid
  GetFeatureStatusNoConfigAndLockStatus' tfn tid -> embedClient $ getFeatureStatusNoConfigAndLockStatus tfn tid
  SetFeatureStatusNoConfig' tfn tid value -> embedClient $ setFeatureStatusNoConfig tfn tid value
```

File: services/galley/test/integration/API/Util/TeamFeature.hs [hunk 2]
```diff
@@ -150,32 +149,50 @@ getAllFeatureConfigsWithGalley galley uid = do
...
+  ( HasCallStack,
+    Public.KnownTeamFeatureName a,
+    ToJSON (Public.TeamFeatureStatus 'Public.WithoutLockStatus a)
+  ) =>
+  (Request -> Request) ->
+  UserId ->
+  TeamId ->
+  Public.TeamFeatureStatus 'Public.WithoutLockStatus a ->
+  TestM ResponseLBS
+putTeamFeatureFlagWithGalley galley uid tid status =
```
Comment (fisx) [inline services/galley/test/integration/API/Util/TeamFeature.hs:163]: you could have implemented this in terms of `putTeamFeatureFlagInternal` below, but we want to get rid of all this code anyway and generate it using servant.  so let's keep this, it certainly looks like it works.

**Review by fisx — APPROVED** [review]

**Comment by fisx** [comment]
last ci failure looks like a flake:

```
brig                             >     sftDiscoveryLoop
brig                             >       when service can be discovered:                                 OK
brig                             >       when service can be discovered and the URLs change:             FAIL
brig                             >         test/unit/Test/Brig/Calling.hs:169:
brig                             >         servers should get overwritten
brig                             >         expected: Discovered (SFTServers (SrvEntry {srvPriority = 0, srvWeight = 0, srvTarget = SrvTarget {srvTargetDomain = "sft4.foo.example.com.", srvTargetPort = 443}} :| [SrvEntry {srvPriority = 0, srvWeight = 0, srvTarget = SrvTarget {srvTargetDomain = "sft5.foo.example.com.", srvTargetPort = 443}}]))
brig                             >          but got: Discovered (SFTServers (SrvEntry {srvPriority = 0, srvWeight = 0, srvTarget = SrvTarget {srvTargetDomain = "sft1.foo.example.com.", srvTargetPort = 443}} :| [SrvEntry {srvPriority = 0, srvWeight = 0, srvTarget = SrvTarget {srvTargetDomain = "sft2.foo.example.com.", srvTargetPort = 443}},SrvEntry {srvPriority = 0, srvWeight = 0, srvTarget = SrvTarget {srvTargetDomain = "sft3.foo.example.com.", srvTargetPort = 443}}]))
```

but i also forgot to push something important.  it's now there.

File: services/galley/test/integration/API/Util/TeamFeature.hs (hunk 2 above)
Comment (battermann) [reply services/galley/test/integration/API/Util/TeamFeature.hs:163]: Hm, but don't we want to test that the authorized team owner can do this? If we use the internal endpoint, that doesn't get tested, instead we test that customer support can change this via the internal API, right?

**Comment by fisx** [comment]
today's not a good day for concourse....

File: services/galley/test/integration/API/Util/TeamFeature.hs (hunk 2 above)
Comment (fisx) [reply services/galley/test/integration/API/Util/TeamFeature.hs:163]: ah, you're right, my bad.

**Pull request merged**